File: src/core/models/balance.py
from sqlalchemy import Column, Integer, String, DateTime, Float, ForeignKey
from core.models.base import Base


class Balance(Base):
    __tablename__ = 'balances'

    id = Column(Integer, primary_key=True)
    accountId = Column(Integer, ForeignKey("accounts.id"), nullable=False)
    available = Column(Float)
    current = Column(Float)
    iso_currency_code = Column(String(8))
    timestamp = Column(DateTime)

    def to_dict(self):
        return {
            'id': self.id,
            'account_id': self.accountId,
            'available': self.available,
            'current': self.current,
            'iso_currency_code': self.iso_currency_code,
            'timestamp': self.timestamp.isoformat(),
        }

File: src/client/src/models/Balance.ts
export interface IBalance {
    id: number;
    accountId: number;
    account_id?: number;
    current: number;
    available: number;
    timestamp: Date;
}

class Balance implements IBalance{
    id: number;
    accountId: number;
    current: number;
    available: number;
    timestamp: Date;

    constructor(serverObj: IBalance = {} as IBalance) {
        this.id = serverObj.id;
        this.accountId = serverObj.account_id;
        this.current = serverObj.current;
        this.available = serverObj.available;
        this.timestamp = serverObj.timestamp;
    }
}

export default Balance;
